airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -236,8 +243,10 @@ private void replicate(final Path jobRoot,
   private static Runnable readFromDstRunnable(final AirbyteDestination destination,
                                               final AtomicBoolean cancelled,
                                               final MessageTracker messageTracker,
+                                              final PersistConfigHelper persistConfigHelper,
                                               final Map<String, String> mdc,
-                                              final ThreadedTimeTracker timeHolder) {
+                                              final ThreadedTimeTracker timeHolder,
+                                              final Long jobId) {
```

Comment:
to avoid adding more parameters to this function, can we make this a part of the PersistConfigHelper constructor?

Why I like this:
- this information is already available when the when the PersistConfigHelper is created,
- the PersistConfigHelper is created per job, so it's ok for it to have job specific information.

Reply:
I decided not to add this as part of the helper constructor because I'd like to use this in the other connector commands like check/discover, which don't have a jobId like this. In that follow-up PR I'll pull out some of the logic into a different method for re-use.